Library/Formula/glassfish.rb

```diff
@@ -6,21 +6,41 @@ class Glassfish < Formula
   url "http://download.java.net/glassfish/4.1/release/glassfish-4.1.zip"
   sha256 "3edc5fc72b8be241a53eae83c22f274479d70e15bdfba7ba2302da5260f23e9d"
 
+  depends_on :java
+  conflicts_with "payara", :because => "Both install `asadmin` binaries"
+
   def install
     rm_rf Dir["bin/*.bat"]
     libexec.install Dir["*", ".org.opensolaris,pkg"]
+    (bin/"asadmin").write <<-EOS.undent
+      #!/bin/sh
+      AS_INSTALL="#{libexec}/glassfish"
+      AS_INSTALL_LIB="$AS_INSTALL/lib"
+      . "${AS_INSTALL}/config/asenv.conf"
+      JAVA=java
+      #Depends upon Java from ../config/asenv.conf
+      if [ ${AS_JAVA} ]; then
+        JAVA=${AS_JAVA}/bin/java
+      fi
+      exec "$JAVA" -jar "$AS_INSTALL_LIB/client/appserver-cli.jar" "$@"
+    EOS
   end
 
   def caveats; <<-EOS.undent
     The home of GlassFish Application Server 4 is:
       #{opt_libexec}
 
-    You may want to add the following to your .bash_profile:
-      export GLASSFISH_HOME=#{opt_libexec}
-      export PATH=${PATH}:${GLASSFISH_HOME}/bin
+    Starting up GlassFish requires root privileges as per
+    https://java.net/jira/browse/GLASSFISH-21343.  The work around is
+    to use
+      sudo asadmin start-domain
 
     Note: The support scripts used by GlassFish Application Server 4
     are *NOT* linked to bin.
   EOS
   end
+
+  test do
+    system "asadmin", "--help"
```

Comment:
`brew audit` doesn't complain here because there is sometimes a reason not to call `bin`, `libexec` or `sbin` on the command, like pipes. There's some API documentation here: http://www.rubydoc.info/github/Homebrew/homebrew/Formula.test